content/data/tanzu-gemfire/blog/command-region-pattern.md

```diff
@@ -0,0 +1,163 @@
+gfsh> start server --name=cluster2-server --locators=localhost[11334] --server-port=40402
+```
+
+* Deploy the *Commands* and *CacheWriter* to both clusters (hint: use the **gfsh deploy** command).
+* Connected to cluster1, create the *region* and the *gateway-sender*.
+
+```
+gfsh> connect --locator=localhost[10334]
+gfsh> create gateway-sender --id=sender1 --remote-distributed-system-id=2
+gfsh> create region --name=distributedCommand --type=REPLICATE_PROXY --gateway-sender-id=sender1 --cache-writer=org.apache.geode.tools.command.internal.DistributedCommandCacheWriter
+```
+
+* Connected to cluster2, create the *region* and the *gateway-receiver*.
+
+```
+gfsh> connect --locator=localhost[11334]
+gfsh> create gateway-receiver
+gfsh> create region --name=distributedCommand --type=REPLICATE_PROXY --cache-writer=org.apache.geode.tools.command.internal.DistributedCommandCacheWriter
+```
+
+* That’s it, you’re ready to start replicating your units of work across clusters!.
+
+## Example
+The following client application simply connects to cluster1 and inserts the *HelloWorldCommand*.
+
+```java
+public class TestClass {
+
+  public static void main(String[] args) {
+    ClientCacheFactory clientCacheFactory = new ClientCacheFactory()
+        .addPoolLocator("localhost", 10334);
+    ClientCache clientCache = clientCacheFactory.create();
+
+    Region<Long, DistributedCommand> region = clientCache
+        .<Long, DistributedCommand>createClientRegionFactory(ClientRegionShortcut.PROXY)
+        .create("distributedCommand");
+
+    region.put(1L, new HelloWorldCommand("TestApplication1"));
+  }
+}
+
+```
+
+It’s easy to see in the logs from both servers (*cluster1-server* and *cluster2-server*) that the command was received and correctly executed by both clusters.
+
+```
+[info 2020/04/10 13:57:24.341 IST <ServerConnection on port 40401 Thread 2> tid=0x51] Executing distributed command HelloWorldCommand...
+[info 2020/04/10 13:57:24.341 IST <ServerConnection on port 40401 Thread 2> tid=0x51] Hello World from TestApplication1!.
+[info 2020/04/10 13:57:24.341 IST <ServerConnection on port 40401 Thread 2> tid=0x51] Executing distributed command HelloWorldCommand... Done!.
+-------------------------------------------------------------------
+[info 2020/04/10 13:57:24.873 IST <ServerConnection on port 5152 Thread 1> tid=0x42] Executing distributed command HelloWorldCommand...
+[info 2020/04/10 13:57:24.873 IST <ServerConnection on port 5152 Thread 1> tid=0x42] Hello World from TestApplication1!.
+[info 2020/04/10 13:57:24.873 IST <ServerConnection on port 5152 Thread 1> tid=0x42] Executing distributed command HelloWorldCommand... Done!.
+```
+
+## What’s Next?
+Check out [geode-command-region-pattern](https://github.com/jujoramos/geode-command-region-pattern) and play around with it, it allows you to test your commands in a distributed fashion using the [geode-dunit](https://cwiki.apache.org/confluence/display/GEODE/DistributedTest) module.
```

Comment:
geode-dunit link "page not found"

Reply:
Sorry.  Page was renamed.  Link should be: https://cwiki.apache.org/confluence/display/GEODE/About+Distributed+Testing